caddyconfig/httpcaddyfile/directives.go

```diff
@@ -62,6 +62,7 @@ var directiveOrder = []string{
 	"push",
 
 	// handlers that typically respond to requests
+	"metrics",
```

Comment:
Are you sure this is the right place? For example, if `redir` is used, that request will never be measured.

Reply:
I'm pretty sure the `metrics` directive is specifically for an HTTP handler to make metrics accessible from arbitrary sites rather than just via the admin API. I don't think this affects metric collection at all (which is always-on after the PR as far as I understand).

Reply:
Oh. Gotcha.

Reply:
That said, I think `metrics` should go after `respond` maybe. `respond` is sometimes used to override access to a certain thing, like:

```
@notLAN not remote_ip 192.168.0.1/16
respond @notLAN "No soup for you" 403

metrics
```

But if `metrics` is first in the order, then I think it'll get sorted before `respond`, making this not work as expected.

Reply:
Hmm... yeah, I had trouble with this, because I was testing with a Caddyfile that had a `respond 404` at the end as a catch-all. It's on my other laptop so I can't paste the exact one right now, but it was something to the effect of:

```
:9180

metrics /metrics

respond /foo "hello world"

respond 404
```

When `respond` was first in the order I got a 404 on `/metrics` which is why I put it here.

I'm open to suggestions though, and the Caddyfile I tested with may not have been relevant 😉

Reply:
I think it's more common for `respond` to be used ahead of other things than as a fallback. The existing directive order optimizes for that right now by having `file_server` after `respond` for example, so I think we should continue doing it that way. Users can always use `route` to override the order if they want to do `respond 404`.

Reply:
Ok, I'll swap `"respond"` and `"metrics"`.